src/transformers/modeling_utils.py

```diff
@@ -2447,6 +2490,17 @@ def save_pretrained(
                 token=token,
             )
 
+    @wraps(PushToHubMixin.push_to_hub)
+    def push_to_hub(self, *args, **kwargs):
+        if "tags" not in kwargs:
+            kwargs["tags"] = self.model_tags
+        elif "tags" in kwargs and self.model_tags is not None:
+            for model_tag in self.model_tags:
+                # merge the tags together
+                if model_tag not in kwargs["tags"]:
+                    kwargs["tags"].append(model_tag)
+        return super().push_to_hub(*args, **kwargs)
```

Comment:
I think your suggestion sounds great, also 
> Is it possible kwargs["tags"] is a string here too?
It can be strings, let me adapt a bit the logic after accepting your suggestion

Reply:
I just took care of the str case in [1e3fc1e](https://github.com/huggingface/transformers/pull/28405/commits/1e3fc1e306bbf953fe513246273fe2544f057288)